expr.go

```diff
@@ -166,9 +166,9 @@ func (l *logicalExpr) String() string {
 // root will return the root of the expr tree
 func root(lExpr *logicalExpr, raw string) (expr, error) {
 	const op = "mql.root"
-	switch {
+	switch lExpr {
 	// intentionally not checking raw, since can be an empty string
-	case lExpr == nil:
+	case nil:
 		return nil, fmt.Errorf("%s: %w (missing expression)", op, ErrInvalidParameter)
 	}
```

Comment:
I actually think this change is probably undesired. This pattern of validating is one we use widely in Boundary. If the linter is suggesting this change, I think it would be better to ignore it or rewrite as an if-statement:
```go
if lExpr == nil {
```

Reply:
Rewrote this as an if statement in 1bcc8ce22574d122b2edfe43a194af3f1c72c1ef.

Reply:
Out of curiosity, why was a switch statement used for a nil check instead of an if statement?